[skip-ci][NFC][ntuple] add missing information to spec (#16704)
root-project/root: tree/ntuple/v7/doc/BinaryFormatSpecification.md

## Glossary

### Anchor

The anchor is a data block that represents the entry point to an RNTuple.
The anchor is specific to the RNTuple container in which the RNTuple data are embedded (e.g., a ROOT file or an object store).
The anchor must provide the information to load the header and the footer **envelopes**.

### Cluster

A cluster is a set of **pages** that contain all the data belonging to an entry range.
The data set is partitioned in clusters.
A typical cluster size is tens to hundreds of megabytes.

### Column

A column is a storage backed vector of a number of **elements** of a simple type.
Column elements have a fixed bit-length that depends on the column type.

> Some column types allow setting the bit lengths within specific limits (e.g. for floats with truncated mantiassa).

Reviewer: ```suggestion
Some column types allow setting the bit lengths within specific limits (e.g. for floats with truncated mantissa).
```

Readers should support reading data written according to _newer_ format versions of the same epoch in the following way:

  - Unknown trailing information in the anchor, in envelopes, and in frames should be ignored.
    For instance, when reading frames, readers should continue reading after the frame-provided frame length
    rather than summing up the lengths of the known contents of the frame.
    Checksum verification, however, should still take place and must include both known and unknown contents.
  - Unknown column, cluster, or field flags should be ignored.
  - Unknown IDs for extra type information should be ignored.
  - When a reader encounters an unknown column type or an unknown field type, field version or field structure,
    it should ignore the entire top-level field this column or field belongs to.
    It should also ignore any projected fields and alias columns whose source fields or columns are already ignored.
  - When a reader encounters an unknown feature flag, it must refuse reading any further.

Writers using format features that will prevent older readers from correctly reading the data
must set the corresponding feature flags.

> Writers should write in the anchor the format version that they support,
> independent of whether they use the all the features that this version provides.
> Only the feature flag signals which features are used.

Reviewer: Upon reading again I think this paragraph can be reworded a bit to improve clarity:

```suggestion
Writers should write in the anchor which format they support,
independent of whether they use the all the features this version provides.
Only the feature flags signal which features are actually implemented by the writer.
```
